ops/storage.py

```diff
@@ -59,11 +60,29 @@ def __init__(self, filename: Union['Path', str]):
             # sqlite3.connect creates the file silently if it does not exist
             logger.debug(f"Initializing SQLite local storage: {filename}.")
 
+        if filename != ":memory:":
+            self._ensure_db_permissions(str(filename))
         self._db = sqlite3.connect(str(filename),
                                    isolation_level=None,
                                    timeout=self.DB_LOCK_TIMEOUT.total_seconds())
         self._setup()
 
+    def _ensure_db_permissions(self, filename: str):
+        """Make sure that the DB file has appropriately secure permissions."""
+        mode = stat.S_IRUSR | stat.S_IWUSR
+        if os.path.exists(filename):
+            try:
+                os.chmod(filename, mode)
+            except OSError as e:
+                logger.warning(f"Unable to adjust permissions of storage file {filename!r}: {e}")
+            return
```

Comment:
Nit: I think this code would be slightly clearer with a blank line after the `return`.